dotCMS/src/main/java/com/dotmarketing/business/web/HostWebAPIImpl.java

```diff
@@ -83,10 +78,19 @@ public Host getCurrentHost(HttpServletRequest request) throws DotDataException,
         }
         
         request.setAttribute(WebKeys.CURRENT_HOST, host);
-        if(session != null && adminMode) {
+        if(session != null && mode.isAdmin) {
             session.setAttribute(WebKeys.CURRENT_HOST, host);
         }
         return host;
 	}
-	
+
+	@Override
+    public Host getCurrentHostNoThrow(HttpServletRequest request) {
+       try {
+           return getCurrentHost(request);
+       }
+       catch(Exception e) {

```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Move this "catch" on the same line that the previous closing curly brace. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3ARightCurlyBraceSameLineAsNextBlockCheck)
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Catch a list of specific exception subtypes instead. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS2221)